QuizApp/View/SearchView/SearchViewController.swift

```diff
@@ -0,0 +1,162 @@
+
+import UIKit
+
+
+class SearchViewController: UIViewController, SearchViewProtocol{
+    
+    //search stack
+    var searchStack = UIStackView()
+    var inputTextField = UITextField()
+    var searchButton = UIButton()
+    
+    var tableView = UITableView()
+    
+    //array kvizova
+    var quizzArray :[Quiz] = []
+    
+    //array kategorija
+    var uniqueSectionArray = [QuizCategory]()
+    
+    //2-d matrica popunjena s kvizovima sortiranim prema kategorijama
+    var matrix = [[Quiz]]()
+    var colorOfCategory =  [UIColor]()
+    
+    var firstLoad = false
+
+    
+    let nService = NetworkService()
+    private let searchViewPresenter = SearchViewPresenter(networking: NetworkService())
+
+    private var router: AppRouter!
+    convenience init(router: AppRouter) {
+        self.init()
+        self.router = router
+      
+    }
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        searchViewPresenter.setSearchViewDelegate(searchViewDelegate: self)
+        
+        //UI
+       buildViews()
+        
+        //skrivanje tipkovnice i odznačivanje textfieldova
+        let tap: UITapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(uiViewPressed))
+        view.addGestureRecognizer(tap)
+        
+        //funkcionalnost
+        searchButton.addTarget(self, action: #selector(searchButtonPressed), for: .touchUpInside)
+        inputTextField.addTarget(self, action: #selector(didTapEmailTextField), for: .editingDidBegin)
+    }
+    
+    
+    
+    
+    
+    
+    @objc func searchButtonPressed(){
+        matrix.removeAll()
+       
+        inputTextField.layer.borderWidth = 0
+        tableView.isHidden = true
+        
+        searchViewPresenter.fetchQuizzesFromCD(with: inputTextField.text! )
+        tableView.reloadData()
+    }
+    
+    @objc private func didTapEmailTextField(){
+        inputTextField.layer.borderWidth = 1
+        inputTextField.layer.borderColor = UIColor.white.cgColor
+       
+    }
+    //odznači inputTextField
+    @objc private func uiViewPressed(){
+        view.endEditing(true)
+        inputTextField.layer.borderWidth = 0
+       
+    }
+    
+    func fetchSuccessful(matrix: [[Quiz]], uniqueSectionArray: [QuizCategory], quizzArray: [Quiz]){
+        
+        self.matrix = matrix
+        self.uniqueSectionArray = uniqueSectionArray
+        self.quizzArray = quizzArray
+        firstLoad = true
+
+        tableView.reloadData()
+        tableView.isHidden = false
+        
+        for j in 0..<uniqueSectionArray.count{
+            
+            let col = UIColor()
+            colorOfCategory.append(col)
+            colorOfCategory[j] = UIColor(red: .random(in: 0...1), green: .random(in: 0...1), blue: .random(in: 0...1), alpha: 1.0)
+        }
+        
+    }
+    
+}
+
+
+
+
+
+extension SearchViewController: UITableViewDelegate{
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
```

Comment:
This func is never called on quiz click so starting quiz from Search does not work.

Reply:
I understand how to fix mistakes in other comments, but this one bothers me still.

I wrote it in the Pull request Readme : "...dok se čudno ponaša na simulatoru(ne želi pokrenuti kviz, ali je responzivan)."

If you run the app on a physical device, SearchVC recognizes a click and opens the quiz, but not from the first try(that's why I wrote it's unresponsive on physical device). I have to fast click few times and than the quiz appears from the searchVC.

Also, is saving data to CD and fetching from CD okay? No comments on that but would like a feedback.